### PR TITLE
Warnings fixed

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: client/common/client.c [hunk 1]
```diff
@@ -1142,7 +1142,7 @@ BOOL client_common_get_access_token(freerdp* instance, const char* request, char
 		goto cleanup;
 	}
 
-	*token = freerdp_utils_aad_get_access_token(log, response, response_length);
+	*token = (const char*)freerdp_utils_aad_get_access_token(log, response, response_length);
```
Comment (pnowack) [inline client/common/client.c]: Is this cast really necessary? `freerdp_utils_aad_get_access_token()` returns `char*` and `*token` also has type `char*`, so this cast looks superfluous.

File: client/common/client.c [hunk 2]
```diff
@@ -1142,7 +1142,7 @@ BOOL client_common_get_access_token(freerdp* instance, const char* request, char
 		goto cleanup;
 	}
 
-	*token = (const char*)freerdp_utils_aad_get_access_token(log, response, response_length);
+	*token = freerdp_utils_aad_get_access_token(log, (const char*)response, response_length);
```
Comment (pnowack) [inline client/common/client.c:1145]: Same here with the token: `freerdp_utils_aad_get_access_token()` returns `char*` and `*token` also has type `char*`, so this cast looks superfluous.

**Review by pnowack — APPROVED** [review]
In the commit `utils/encoded_types: Get rid of implicit conversion warnings`, the `Get rid of these warnings by adding dummy shifting operations.` part can now be replaced with the following (since the content of the commit changed here regarding to what it does):
```
Get rid of these warnings by explicitly casting the respective values
before shifting them.
```

File: client/common/client.c (hunk 2 above)
Comment (akallabeth) [reply client/common/client.c:1145]: I think that comment is for something before a browser refresh?

File: client/common/client.c (hunk 1 above)
Comment (akallabeth) [reply client/common/client.c]: looks like a comment for an old version?

File: client/common/client.c (hunk 2 above)
Comment (pnowack) [reply client/common/client.c:1145]: > I think that comment is for something before a browser refresh?

yes, indeed.

**Review by pnowack — APPROVED** [review]

**Pull request merged**